### DISCUSSION TITLE
Build for ios with air on linux

### DISCUSSION THREAD

**Discussion by ladigitale** [airsdk/Adobe-Runtime-Support#3547] [Q&A]
I've just switched from windows to linux.
No problem building for android, but when I build for ios the AIr SDK gives me this error: 

“SDK is missing file MainWindow.nib."

Does anyone have this kind of problem?

Thanks a lot.

**Comment by ajwfrost** [comment]
We don't support packaging for iOS on a Linux machine. Packaging up requires a number of separate executable files - in the SDK under /lib/aot/bin - and we don't have Linux variants of those currently..

That said .. it may be possible to get this working. Certainly if you considered the "remote linking on mac" feature, which would mean we don't need an "ld" linker binary, which is the tricky one.

The other tools:
- as - not actually sure whether this is ever used..
- compile-abc - we build this ourselves for Windows and macOS. Theoretically possible to build this for Linux too I think.
- iOSBin - a mixture of home-created utilities plus a third party (ideviceinfo) which I think is available for Linux
- isb - wrapper around iOSBin iirc
- ld64 - the main problem .. although now, for Windows this is actually the LLVM linker, so we may be able to also get that building for Linux, with the same caveats that it's unlikely to work properly if you're using certain ANEs.
- lipo - we created a basic version of this mac tool, trivial
- strip - don't think this is used either.

So it might be possible to do this, if people really wanted to...

**Comment by ladigitale** [comment]
Thank you for the answer
